remotespark/utils/utils.py

```diff
@@ -97,3 +99,21 @@ def get_instance_id():
 
     return instance_id
 
+
+def coerce_pandas_df_to_numeric_datetime(df):
+    for column_name in df.columns:
+        coerced = False
+
+        if not coerced and df[column_name].dtype == np.dtype("object"):
+            try:
+                df[column_name] = pd.to_datetime(df[column_name], errors="raise")
+                coerced = True
+            except:
```

Comment:
Catching all exceptions is a bad idea.  Identify the actual exception that Pandas throws when a coercion error happens and catch only those.  If the exception Pandas throws is unpredictable or unknown, then at least `except Exception:` -- but that should be a last-resort thing